### PR TITLE
feat(user-profile): UserProfile Email Notification & gql refactor

### PR DESCRIPTION
**Pull request by GunnlaugurG**
## What

* Added emaiNotification field to the userProfile table. EmaiNotification controls if user can receive emails from island.is. 
* Added userprofile v2 to the graphql layer and added service that decides if  V1 or V2 should be called based on a feature flag

## Why

* Since we are taking steps to remove `islyklarService` we will need to keep canNudge in our database. CanNudge property was renamed to Email Notification
* When the UserProfile V2 is enabled we will need all services that use UserProfile to call V2, thats why I added the graphql service that decides weather to call V1 or V2 based on the feature flag controlling UserProfile V2

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/user-profile/src/lib/V1/userProfile.service.ts
Comment (saevarma) [inline libs/api/domains/user-profile/src/lib/V1/userProfile.service.ts:1]: `libs/api/domains/user-profile/src/lib/v1/userProfile.service.ts` folder name should use small `v`

File: libs/api/domains/user-profile/src/lib/V1/userProfile.service.ts [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    category: LOG_CATEGORY,
+  })
+  throw new ApolloError('Failed to resolve request', error.status)
+}
+
+@Injectable()
+export class UserProfileServiceV1 {
+  constructor(
+    private userProfileApi: UserProfileApi,
+    private v2MeApi: V2MeApi,
```
Comment (saevarma) [inline libs/api/domains/user-profile/src/lib/V1/userProfile.service.ts:44]: Unused

File: libs/api/domains/user-profile/src/lib/V2/userProfile.service.ts
Comment (saevarma) [inline libs/api/domains/user-profile/src/lib/V2/userProfile.service.ts:1]: `libs/api/domains/user-profile/src/lib/v2/userProfile.service.ts`

File: libs/api/domains/user-profile/src/lib/V2/userProfile.service.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import { IslykillService } from '../islykill.service'
+import { UserProfile } from '../userProfile.model'
+import { UpdateUserProfileInput } from '../dto/updateUserProfileInput'
+import { CreateSmsVerificationInput } from '../dto/createSmsVerificationInput'
+import { CreateEmailVerificationInput } from '../dto/createEmalVerificationInput'
+import { ConfirmSmsVerificationInput } from '../dto/confirmSmsVerificationInput'
+import { ConfirmEmailVerificationInput } from '../dto/confirmEmailVerificationInput'
+
+/** Category to attach each log message to */
+const LOG_CATEGORY = 'userprofile-service'
```
Comment (saevarma) [inline libs/api/domains/user-profile/src/lib/V2/userProfile.service.ts:19]:
```suggestion
const LOG_CATEGORY = 'userprofile-service-v2'
```

File: libs/api/domains/user-profile/src/lib/userProfile.service.ts [hunk 3]
```diff
@@ -44,39 +43,27 @@ export class UserProfileService {
...
   userProfileApiWithAuth(auth: Auth) {
     return this.userProfileApi.withMiddleware(new AuthMiddleware(auth))
   }
 
-  async getIslykillProfile(user: User) {
-    try {
-      const islyklarData = await this.islyklarService.getIslykillSettings(
-        user.nationalId,
-      )
+  private async services(user: User) {
```
Comment (saevarma) [inline libs/api/domains/user-profile/src/lib/userProfile.service.ts:55]:
```suggestion
  private async getService(user: User) {
```

File: libs/api/domains/user-profile/src/lib/userProfile.service.ts [hunk 4]
```diff
@@ -90,187 +77,27 @@ export class UserProfileService {
...
   async getUserProfile(user: User) {
-    try {
-      const profile = await handle204(
-        this.userProfileApiWithAuth(
-          user,
-        ).userProfileControllerFindOneByNationalIdRaw({
-          nationalId: user.nationalId,
-        }),
-      )
+    const service = await this.services(user)
```
Comment (saevarma) [inline libs/api/domains/user-profile/src/lib/userProfile.service.ts:80]:
```suggestion
    const service = await this.getService(user)
```

File: libs/service-portal/graphql/src/hooks/profile/useUserProfile.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { useQuery } from '@apollo/client'
+import { ApolloError, useLazyQuery, useQuery } from '@apollo/client'
```
Comment (saevarma) [inline libs/service-portal/graphql/src/hooks/profile/useUserProfile.ts:1]: Unused imports

File: libs/service-portal/graphql/src/lib/queries/getUserProfile.ts [hunk 6]
```diff
@@ -12,6 +12,7 @@ export const USER_PROFILE = gql`
       emailStatus
       mobileStatus
       modified
+      emailNotifications
```
Comment (saevarma) [inline libs/service-portal/graphql/src/lib/queries/getUserProfile.ts:15]: Use this instead and remove `canNudge` from the query.

**Review by saevarma — APPROVED** [review]
Great job 🙌

**Review by thordurhhh — APPROVED** [review]
LGTM

**Pull request merged**